Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -301,36 +309,84 @@ extension Calendar {
                     return
                 }
                 
+                let calendar = recurrence.calendar
+                                
                 var dates: [Date] = [anchor]
-                 
+                     
+                let components = calendar._dateComponents([.second, .minute, .hour, .day, .month, .isLeapMonth, .dayOfYear, .weekday], from: anchor) 
+
+                var componentCombinations = Calendar._DateComponentsCombinations()
+                
+                if recurrence.frequency == .yearly || recurrence.frequency == .monthly {
+                    if recurrence.daysOfTheYear.isEmpty {
+                        componentCombinations.months = if recurrence.months.isEmpty { [RecurrenceRule.Month(from: components)!] } else { recurrence.months }
+                        componentCombinations.daysOfMonth = if recurrence.daysOfTheMonth.isEmpty { [components.day!] } else { recurrence.daysOfTheMonth}
+                        componentCombinations.daysOfYear = nil
+                    } else {
+                        componentCombinations.months = nil
+                        componentCombinations.daysOfMonth = nil
+                        componentCombinations.daysOfYear = if recurrence.daysOfTheYear.isEmpty { [components.day!] } else { recurrence.daysOfTheYear}
+                    }
+                } else {
+                    componentCombinations.months = nil
+                    componentCombinations.daysOfMonth = nil
+                    componentCombinations.daysOfYear = nil
+                }
+                let searchInterval = calendar.dateInterval(of: recurrence.frequency.component, for: anchor)!
+                let searchRange = searchInterval.start..<searchInterval.end
+                let searchStart = searchInterval.start
+                if weekdayAction == .expand {
+                    componentCombinations.weekdays = recurrence.weekdays
+                    componentCombinations.daysOfYear = nil
+                    componentCombinations.daysOfMonth = nil
+                } else if recurrence.frequency == .weekly || !recurrence.weeks.isEmpty {
+                   if let weekdayIdx = components.weekday, let weekday = Locale.Weekday(weekdayIdx) {
+                       componentCombinations.weekdays = [.every(weekday)]
+                       componentCombinations.daysOfYear = nil
+                       componentCombinations.daysOfMonth = nil
+                    }
+                }
+                if recurrence.frequency != .hourly, recurrence.frequency != .minutely {
+                    componentCombinations.hours   = if hourAction   == .expand { recurrence.hours   } else { components.hour.map { [$0] } }
```

Comment:
Also I knew we don't have a linter yet, but can you still clean up the whitespaces while you're at it?